iree-prof-tools/graph-util.cc

```diff
@@ -464,11 +586,46 @@ void AddNode(
   AddOutputMetadata(node);
 }
 
+// Append another namespace to group the operations if same operations continue
+// too many times in a row.
+void AppendNamespacesIfNecessary(const Graph& graph,
```

Comment:
We may come up with a better algorithm here based on the compiler knowledge.

For example, we can group a series of nodes with a single output nodes, since they represent sequential operations. There might be more grouping possible for a subgraph, and it would need a compiler's help.